File: src/ts/app/Author.ts
class Author {
    private readonly _id: string;
    private readonly _name: string;
    private readonly _avatarUrl: string;

    constructor(json: any) {
        this._id = json.id;
        this._name = json.name;
        this._avatarUrl = json.avatarUrl;
    }

    get id(): string {
        return this._id;
    }

    get name(): string {
        return this._name;
    }

    get avatarUrl(): string {
        return this._avatarUrl;
    }
}

export default Author;

File: OliverBooth.Api/Data/Author.cs
using System.Text.Json.Serialization;
using OliverBooth.Common.Data.Blog;
using OliverBooth.Common.Data.Web.Users;

namespace OliverBooth.Api.Data;

internal sealed class Author
{
    [JsonPropertyName("avatarUrl"), JsonInclude, JsonPropertyOrder(2)]
    public Uri AvatarUrl { get; private set; } = null!;

    [JsonPropertyName("id"), JsonInclude, JsonPropertyOrder(0)]
    public Guid Id { get; private set; }

    [JsonPropertyName("name"), JsonInclude, JsonPropertyOrder(1)]
    public string Name { get; private set; } = string.Empty;

    public static Author FromUser(IUser author)
    {
        return new Author
        {
            Id = author.Id,
            Name = author.DisplayName,
            AvatarUrl = author.AvatarUrl
        };
    }
}
